### PR TITLE
Added obsolete fields: requires, provides and obsoletes

### PR DESCRIPTION
**Pull request by venthur**
closes: #1107

### CODE REVIEW

**Review by Harlee1994 — APPROVED** [review]

File: source/specifications/core-metadata.rst [hunk 1]
```diff
@@ -755,6 +755,92 @@ Examples::
...
+
+Deprecated Fields
+=================
+
+Requires
+--------
+
+.. versionadded:: 1.1
+.. deprecated:: 1.2
+   in favour of ``Requires-Dist``
```
Comment (bhrutledge) [inline source/specifications/core-metadata.rst:767]: Given that all of these are deprecated fields, what value do the detailed descriptions provide? At first glance, it seems like they mostly duplicate the current fields. Could we just limit it to these first 5 lines per section?

Comment (venthur) [reply source/specifications/core-metadata.rst:767]: They are deprecated but still work, see https://github.com/pypa/packaging.python.org/issues/1107 -- so it makes sense to document them properly.

File: source/specifications/core-metadata.rst [hunk 2]
```diff
@@ -755,6 +755,92 @@ Examples::
...
+
+Each entry contains a string describing some other module or package required
+by this package.
+
+The format of a requirement string is identical to that of a module or package
+name usable with the ‘import’ statement, optionally followed by a version
```
Comment (bhrutledge) [inline source/specifications/core-metadata.rst:773]: Nit: let's use monospace for this:

```suggestion
name usable with the ``import`` statement, optionally followed by a version
```

File: source/specifications/core-metadata.rst (hunk 1 above)
Comment (bhrutledge) [reply source/specifications/core-metadata.rst:767]: @dstufft Is this what you had in mind when you submitted #1107? Are the descriptions accurate/helpful?

Comment (dstufft) [reply source/specifications/core-metadata.rst:764]: Sorry, I missed this PR somehow.

I don't personally feel strongly, though the "Rarely Used" fields are probably the closest analogy to these and they have their full documentation, so I guess I'd lean towards including it?

**Review by dstufft — APPROVED** [review]

**Comment by venthur** [comment]
@webknjaz the linkcheck is failing and preventing the auto-merge. I didn't add any links in this PR and I also can't figure out based on its output what it is complaining about.

File: source/key_projects.rst [hunk 3]
```diff
@@ -558,7 +558,7 @@ resolution by locally caching metadata about dependencies.
 pypiserver
 ==========
 
-`Docs <https://github.com/pypiserver/pypiserver/blob/master/README.rst>`__ |
+`Docs <https://github.com/pypiserver/pypiserver/blob/master/README.md>`__ |
```
Comment (webknjaz) [inline source/key_projects.rst:561]: I'd rather have this as https://github.com/pypiserver/pypiserver#readme instead. Though, it's probably best to fix it in a dedicated PR.

Comment (venthur) [reply source/key_projects.rst:561]: I assume, nothing to do here then? I'm also not happy to fix unrelated thing in this PR, but the CI pipeline is set up in a way that prevented the auto merge when a link is broken, even if the link was not introduced by the PR. 

Maybe this setup should be addressed as well at some point, could be confusing for some contributors.

**Comment by chrysle** [comment]
@venthur Could you rebase? The problem has been fixed recently.

**Comment by venthur** [comment]
fixed, but now linkcheck fails due to a timeout of another url. this test is a bit fragile. can someone trigger a re-run of the pipeline? i think i can't do it myself.

**Pull request merged**